drivers/hwinfo/Kconfig

```diff
@@ -70,12 +70,20 @@ config HWINFO_IMXRT
 	help
 	  Enable NXP i.mx RT hwinfo driver.
 
+config HWINFO_SAM_RSTC
+	bool "Atmel SAM reset cause"
+	default y
+	depends on SOC_SERIES_SAM4S || SOC_SERIES_SAME70 || \
+		   SOC_SERIES_SAMV71
```

Comment:
I think only SoC version that not uses this driver is SAM4L.

```diff
-	depends on SOC_SERIES_SAM4S || SOC_SERIES_SAME70 || \
-		   SOC_SERIES_SAMV71
+	depends on SOC_FAMILY_SAM && !SOC_SERIES_SAM4L
`````

Reply:
The other SoCs are missing defines in the RSTC header file, so the driver didn’t compile for those.

Reply:
Hi @pdgendt ,
If you don't mind, could you fix it at hal_atmel? This is not a block, it is just to get the complete feature on all supported SoC.

Reply:
I’m currently on vacation without a development setup. It can be a separate PR, or it’ll have to wait. 😅

Reply:
No problem, fell free to update in the future. It is a nice to have.